### PR TITLE
feat: 🎸 add `onStatusChange` to confidential transaction

### PR DESCRIPTION
**Pull request by polymath-eric**
### Description

allows for subscription to confidential transaction status changes

### Breaking Changes

none

### JIRA Link

✅ Closes: [DA-995](https://polymesh.atlassian.net/browse/DA-995)

### Checklist

- [ ] Updated the Readme.md (if required) ?

[DA-995]: https://polymesh.atlassian.net/browse/DA-995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts [hunk 1]
```diff
@@ -149,6 +152,85 @@ describe('ConfidentialTransaction class', () => {
     });
   });
 
+  describe('method: onStatusChange', () => {
+    let bigNumberToU64Spy: jest.SpyInstance;
+    let instructionStatusesMock: jest.Mock;
+    const rawId = dsMockUtils.createMockU64(new BigNumber(1));
```
Comment (prashantasdeveloper) [inline src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts:158]: nit -> I prefer initializing in the beforeEach/beforeAll section especially the mock values

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts [hunk 2]
```diff
@@ -149,6 +152,85 @@ describe('ConfidentialTransaction class', () => {
...
+    beforeAll(() => {
+      bigNumberToU64Spy = jest.spyOn(utilsConversionModule, 'bigNumberToU64');
+    });
+
+    beforeEach(() => {
+      const owner = 'someDid';
+      entityMockUtils.configureMocks({ identityOptions: { did: owner } });
+      when(bigNumberToU64Spy).calledWith(id, context).mockReturnValue(rawId);
+
+      instructionStatusesMock = dsMockUtils.createQueryMock(
```
Comment (prashantasdeveloper) [inline src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts:173]:
```suggestion
      transactionStatusesMock = dsMockUtils.createQueryMock(
```
nit -> feels like got left over from copy paste

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

File: src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts (hunk 2 above)
Comment (polymath-eric) [reply src/api/entities/confidential/__tests__/ConfidentialTransaction/index.ts]: 😅

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**